tui/network_dialog_check.go

```diff
@@ -0,0 +1,216 @@
+func (dialog *NetworkTestDialog) LoopWaitDuration() time.Duration {
+	return 1 * time.Second
+}
+
+// OnClose sets the callback that is called when the
+// dialog is closed
+func (dialog *NetworkTestDialog) OnClose(fn func()) {
+	clui.WindowManager().BeginUpdate()
+	defer clui.WindowManager().EndUpdate()
+	dialog.onClose = fn
+}
+
+// Close closes the dialog window and executes a callback if registered
+func (dialog *NetworkTestDialog) Close() {
+	clui.WindowManager().DestroyWindow(dialog.DialogBox)
+	clui.WindowManager().BeginUpdate()
+	closeFn := dialog.onClose
+	_ = term.Flush() // This might be dropped once clui is fixed
+	clui.WindowManager().EndUpdate()
+	if closeFn != nil {
+		closeFn()
+	}
+}
+
+func initDiaglogWindow(dialog *NetworkTestDialog) error {
+
+	const title = "Testing Networking..."
+	const wBuff = 5
+	const hBuff = 4
+	const dWidth = 50
+	const dHeight = 8
+
+	sw, sh := clui.ScreenSize()
+
+	x := (sw - TermWidth) / 2
+	y := (sh - TermHeight) / 2
+
+	posX := (TermWidth - dWidth + wBuff) / 2
+	if posX < wBuff {
+		posX = wBuff
+	}
+	posX = x + posX
+	posY := (TermHeight-dHeight+hBuff)/2 - hBuff
+	if posY < hBuff {
+		posY = hBuff
+	}
+	posY = y + posY
+
+	dialog.DialogBox = clui.AddWindow(posX, posY, dWidth, dHeight, title)
+	dialog.DialogBox.SetTitleButtons(0)
+	dialog.DialogBox.SetMovable(false)
+	dialog.DialogBox.SetSizable(false)
+	clui.WindowManager().BeginUpdate()
+	defer clui.WindowManager().EndUpdate()
+	dialog.DialogBox.SetModal(true)
+	dialog.DialogBox.SetConstraints(dWidth, dHeight)
+	dialog.DialogBox.SetPack(clui.Vertical)
+
+	borderFrame := clui.CreateFrame(dialog.DialogBox, dWidth, dHeight, clui.BorderNone, clui.Fixed)
+	borderFrame.SetPack(clui.Vertical)
+	borderFrame.SetGaps(0, 1)
+	borderFrame.SetPaddings(1, 1)
+
+	dialog.progressBar = clui.CreateProgressBar(borderFrame, AutoSize, 1, clui.Fixed)
+	_, dialog.progressMax = dialog.progressBar.Limits()
+
+	dialog.resultLabel = clui.CreateLabel(borderFrame, 1, 1, "Connecting to the network servers...", 1)
+	dialog.resultLabel.SetMultiline(true)
+
+	buttonFrame := clui.CreateFrame(borderFrame, AutoSize, 1, clui.BorderNone, clui.Fixed)
+	buttonFrame.SetPack(clui.Horizontal)
+	buttonFrame.SetGaps(1, 0)
+	dialog.dismissButton = CreateSimpleButton(buttonFrame, AutoSize, AutoSize, "Dismiss", Fixed)
+	dialog.dismissButton.SetEnabled(false)
+
+	return nil
+}
+
+// CreateNetworkTestDialogBox creates the Network PopUp
+func CreateNetworkTestDialogBox(modelSI *model.SystemInstall) (*NetworkTestDialog, error) {
+	dialog := new(NetworkTestDialog)
+
+	if dialog == nil {
+		return nil, fmt.Errorf("Failed to allocate a Network Test Dialog")
+	}
+
+	if err := initDiaglogWindow(dialog); err != nil {
+		return nil, fmt.Errorf("Failed to create Network Test Dialog: %v", err)
+	}
+
+	if modelSI == nil {
+		return nil, fmt.Errorf("Missing model for Network Test Dialog")
+	}
+	dialog.modelSI = modelSI
+
+	dialog.dismissButton.OnClick(func(ev clui.Event) {
+		go func() {
```

Comment:
you don't need this to be in a separate go routine

Reply:
Good catch. Removed.